src/cysignals/implementation.c

```diff
@@ -438,18 +441,26 @@ static void _sig_off_warning(const char* file, int line)
     print_backtrace();
 }
 
-
 static void setup_alt_stack(void)
 {
 #if HAVE_SIGALTSTACK
-    /* Static space for the alternate signal stack. The size should be
+    /* Space for the alternate signal stack. The size should be
      * of the form MINSIGSTKSZ + constant. The constant is chosen rather
      * ad hoc but sufficiently large. */
-    static char alt_stack[MINSIGSTKSZ + 5120 + BACKTRACELEN * sizeof(void*)];
-
     stack_t ss;
+#if HAVE_SYS_MMAN_H
+    /* MISIGSTKSZ is no longer a constant starting with glibc 2.34.
+     * If <sys/mmap.h> is present, we can use mmap as a workaround as
+     * done in
+     * https://github.com/rr-debugger/rr/commit/a744ceff00ffc04c9b3e9cc9d424d434c5a808b7
+     * */
+    size_t stack_size = MINSIGSTKSZ + 5120 + BACKTRACELEN * sizeof(void*);
+    ss.ss_sp = mmap(NULL, stack_size, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
+    ss.ss_size = stack_size;
+#else
     ss.ss_sp = alt_stack;
     ss.ss_size = sizeof(alt_stack);
```

Comment:
The non-mmap case is broken now since you deleted the statically allocated `alt_stack` buffer.

I would suggest adding a configure check for whether or not `MINSIGSTKSZ` is constant, *however* if this is only going to be the case for older glibc versions moving forward, there's probably not much advantage in keeping backwards compat for the static allocation, since this is just code run at setup time.

Instead, as a fallback, we could use malloc as in the example given here: https://man7.org/linux/man-pages/man2/sigaltstack.2.html

In both cases (malloc and mmap) there should be error checking in case the allocation failed.

I'm not exactly sure where the constant 5120 comes from.  It's 5kb, which is one (typical) page size on Linux plus an extra kb (I'm guessing just as a safety margin).  It might also be that OSX requires a larger buffer, which I've seen referenced somewhere.

If we're getting rid of the static allocation we can also replace this constant with something like `sysconf(_SC_PAGESIZE) * 2` which should be sufficient.

I also like [this suggestion](https://stackoverflow.com/a/31820184/982257) of, when using `mmap`, to use `mprotect()` to create a couple extra guard pages on either end of the allocation (but use `sysconf(_SC_PAGESIZE)` instead of `getpagesize()`.

Another concern we need to look into: According to this commit ee6e81eaed82f87690da21be4ef7d3bc7b298473 we used to malloc the buffer, but that had problems on some older (?) OSX versions.  Jeroen wasn't totally clear about what the issue was (and without blame--it sounds confusing), but it apparently had something to do with fork: https://trac.sagemath.org/ticket/21509#comment:21

So actually, maybe for the sake of OSX, the static allocation should be kept as a possibility too.  That or the OSX issue is fixed upstream, and we don't need to care about older versions since Apple enforces planned obsolescence.  Still, someone with a more recent Mac should check this.

Reply:
I resolved the non-mmap case now.

I'm completely uncertain what the best solution would be. `malloc` sounds nice and easy. `_setup_alt_stack` isn't even used as far as I can see other than in cysignals doctests. I don't know why `malloc` would not work in this case.

There is a small memory leak when calling `init_cysignals` more than once. But maybe this is fine? But could also be avoided by storing the address somewhere in `implementation.c`, so that on reinitialization it is being recycled.